packages/request-handler/src/stock-price.ts

```diff
@@ -11,13 +12,27 @@ export const fetchStockPriceHandler = createCachedWidgetRequestHandler({
     const response = await fetchWithTimeout(
       `https://query1.finance.yahoo.com/v8/finance/chart/${input.stock}?range=${input.timeRange}&interval=${input.timeInterval}`,
     );
-    const data = await response.json();
+    const data = await response.json() as object;
 
     if (data.error) {
       throw new Error(data.error.description);
     }
 
-    return data.chart.result[0];
+    return responseSchema.parse(data.chart.result[0]);
```

Comment:
I would suggest to parse `data` the `await response.json()` directly.
So you would probably need something in the direction of

```
z.object({
	chart: z.object({
		result: z.array(responseSchema)
	})
}).or(z.object({
	error: z.object({
		description: z.string()
	})
}))
```